src/input/Search.js

```diff
@@ -132,6 +137,20 @@ class Search extends Component {
           )
         }
         {
+          clearIcon && (
+            <Icon
+              size={16}
+              style={[
+                styles.clearIcon,
+                clearIcon.style && clearIcon.style
+              ]}
+              name={clearIcon.name || 'close'}
+              onPress={this.clear.bind(this)}
```

Comment:
rename function to make intent more explicit? Something like: `clearText` would work too.